common/util.go

```diff
@@ -73,12 +73,12 @@ const (
 	FailureReasonFailureDetailsExceedsLimit = "FAILURE_DETAILS_EXCEEDS_LIMIT"
 	// FailureReasonCancelDetailsExceedsLimit is failureReason for cancel details exceeds limit
 	FailureReasonCancelDetailsExceedsLimit = "CANCEL_DETAILS_EXCEEDS_LIMIT"
-	//FailureReasonHeartbeatExceedsLimit is failureReason for heartbeat exceeds limit
+	// FailureReasonHeartbeatExceedsLimit is failureReason for heartbeat exceeds limit
 	FailureReasonHeartbeatExceedsLimit = "HEARTBEAT_EXCEEDS_LIMIT"
 	// FailureReasonDecisionBlobSizeExceedsLimit is the failureReason for decision blob exceeds size limit
 	FailureReasonDecisionBlobSizeExceedsLimit = "DECISION_BLOB_SIZE_EXCEEDS_LIMIT"
-	// TerminateReasonSizeExceedsLimit is reason to terminate workflow when history size or count exceed limit
-	TerminateReasonSizeExceedsLimit = "HISTORY_EXCEEDS_LIMIT"
+	// FailureReasonSizeExceedsLimit is reason to terminate workflow when history size or count exceed limit
```

Comment:
Why change to fail instead of terminate? I thought terminate is cleaner and more proper in this case, since customer may not expect or handle workflows terminate state.

Reply:
BTW comment not changed.

Reply:
because existing logic fails a workflow in decision handling:
https://github.com/uber/cadence/blob/master/service/history/decisionChecker.go#L113

the reason to move the size / count check here is
1. decision is inevitable
2. workflow execution context has too much "business logic"